x-pack/plugins/alerting/server/rules_client/lib/apply_bulk_edit_operation.ts

```diff
@@ -49,5 +51,8 @@ export const applyBulkEditOperation = <R extends object>(operation: BulkEditOper
       break;
   }
 
-  return rule;
+  return {
+    modifiedAttributes: rule,
+    isAttributeModified: !isEqual(originalFieldValue, get(rule, operation.field)),
```

Comment:
Added tests for `isAttributeModified: false` but only to `tags` operation for now: `schedule` and `timeline` operations have no logic for skipping now (should they?)

Also wondering about `actions`: right now a user can add an already existing action, but it will be a different object per se, so the deduplication we do with `Set` won't work. Should we allow the same action to be added? (I'm thinking especially about doing it via the API)

Also if the user `set`s the actions array to be exactly the same as the existing, should we consider that to be `skipped`?

Reply:
> Added tests for isAttributeModified: false but only to tags operation for now

That sounds good to me, thanks @jpdjere 👍 

> schedule and timeline operations have no logic for skipping now (should they?)

Not sure what do you mean here

> Also wondering about actions: right now a user can add an already existing action, but it will be a different object per se, so the deduplication we do with Set won't work. Should we allow the same action to be added? (I'm thinking especially about doing it via the API)

That's a good question. I think allowing that is not a big deal. We could probably disallow but it would mean we'd need to handle this restriction in the UI as well -- I would avoid working on this for now.
@jpdjere let's double-check this with Xavier.

> Also if the user sets the actions array to be exactly the same as the existing, should we consider that to be skipped?

Yes, we should treat this operation as skipped in this case. This is what my question was about: can there be any edge cases that wouldn't work with `isAttributeModified: !isEqual(originalFieldValue, get(rule, operation.field))`?

This is what an action is:

```ts
export type RuleActionParams = SavedObjectAttributes;
export type RuleActionParam = SavedObjectAttribute;

export interface RuleAction {
  group: string;
  id: string;
  actionTypeId: string;
  params: RuleActionParams;
  frequency?: {
    summary: boolean;
    notifyWhen: RuleNotifyWhenType;
    throttle: string | null;
  };
}
```

`SavedObjectAttributes` can be anything and can include nested objects and arrays. I'm wondering how this equality check would work for arbitrarily nested objects.